File: application/core/src/main/java/br/com/beblue/bluerecords/core/usecase/DiscoUseCase.java
package br.com.beblue.bluerecords.core.usecase;

import br.com.beblue.bluerecords.core.command.CadastraDiscoCommand;
import br.com.beblue.bluerecords.core.command.ConsultaDiscoCommand;
import br.com.beblue.bluerecords.core.entitidade.Disco;
import br.com.beblue.bluerecords.core.entitidade.Genero;
import br.com.beblue.bluerecords.core.entitidade.paginacao.Paginacao;

import java.util.List;

public interface DiscoUseCase {
    Disco consultar(Integer id);
    Boolean discosForamCadastrados();
    List<CadastraDiscoCommand> consultarDiscosParaCadastroPorGenero(Genero genero);
    Paginacao<Disco> consultarDiscos(ConsultaDiscoCommand command);
    Disco cadastrar(CadastraDiscoCommand command);
}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/usecase/disco/DiscoUseCase.kt
package br.com.beblue.bluerecords.core.usecase.disco

import br.com.beblue.bluerecords.core.entity.Disco
import br.com.beblue.bluerecords.core.entity.Genero
import br.com.beblue.bluerecords.core.usecase.disco.command.CadastrarDiscoCommand

interface DiscoUseCase
{
    fun consultarDiscoPorId(id: Int): Disco
    fun discosForamCadastrados() : Boolean
    fun consultarDiscosParaCadastroPorGenero(genero: Genero): List<CadastrarDiscoCommand>
    fun cadastrarDisco(command: CadastrarDiscoCommand) : Disco
}